Add access control to components implementing ListContext (#10188)
marmelab/react-admin: docs/InfiniteList.md

);
```

## `unauthorized`

The component to display when users don't have access to the list:

```tsx
import { CreateButton, InfiniteList } from 'react-admin';
import { Box, Button, Typography } from '@mui/material';

const Unauthorized = () => (
    <Box textAlign="center" m={1}>
        <Typography variant="h4" paragraph>
            You don't have access to this list
        </Typography>
        <Typography variant="body1">
            Contact the administrator to request access
        </Typography>
        <CreateButton />

>         <Button onClick={...}>Import</Button>

Reviewer: I don't understand what that button does here. Same for the other components.

>         <Button onClick={...}>Import</Button>

Author: ```suggestion
```

Bad copy/paste from the empty example

};
```

`useInfiniteListController` returns callbacks to sort, filter, and paginate the list, so you can build a complete List page.

## Access Control

> Should your authProvider implement the [`canAccess` method](./AuthProviderWriting.md#canaccess), the `<InfiniteList>` component will call it to ensure users have the right to access to its data. For instance, given the following `<List>`: 

Author: ```suggestion
Should your authProvider implement the [`canAccess` method](./AuthProviderWriting.md#canaccess), the `<InfiniteList>` component will call it to ensure users have the right to access to its data. For instance, given the following `<InfiniteList>`: 
```

};
```

`useInfiniteListController` returns callbacks to sort, filter, and paginate the list, so you can build a complete List page.

> ## Access Control

Reviewer: I rephrased the doc for this component, and I think you should do the same for the others

> Should your authProvider implement the [`canAccess` method](./AuthProviderWriting.md#canaccess), the `<InfiniteList>` component will call it to ensure users have the right to access to its data. For instance, given the following `<InfiniteList>`: 

Reviewer: ```suggestion
Should your authProvider implement the [`canAccess` method](./AuthProviderWriting.md#canaccess), the `<InfiniteList>` component will call it to only display data the current user has the right to access.
 
### Page Access

`<InfiniteList>` only renders if the user has the right to display the page (`{ action: "read", resource: "[resourceName]" }`).

For instance, given the following `<InfiniteList>`: 
```
```suggestion
Should your authProvider implement the [`canAccess` method](./AuthProviderWriting.md#canaccess), the `<InfiniteList>` component will call it to ensure users have the right to access to its data. For instance, given the following `<InfiniteList>`: 
```

```tsx
import { InfiniteList, Datagrid, TextField } from 'react-admin';

// Resource name is "posts"
const PostList = () => (
    <InfiniteList>
        <Datagrid>
            <TextField source="title" />
            <TextField source="author" />
            <TextField source="published_at" />
        </Datagrid>
    </InfiniteList>
);
```

> The `<InfiniteList>` will call `authProvider.canAccess()` using the following parameters: `{ action: "read", resource: "posts" }`.

Reviewer: ```suggestion
The `<InfiniteList>` will call `authProvider.canAccess()` using the following parameters: 

- `{ action: "read", resource: "posts" }`.
```

**Tip**: If you want to control what columns users can see in the `<Datagrid>`, leverage [its access control feature](./Datagrid.md#access-control).

### Exported Fields

Should your authProvider implement the [`canAccess` method](./AuthProviderWriting.md#canaccess), the [`exporter`](#exporter) hook will call it to export only fields the current user has the right to export. For instance, given the following record shape:

```json
{
    "id": 0,
    "title": "Never Gonna Give You Up",
    "author": "Rick Astley",
}
```

> The `authProvider.canAccess` method will be called 3 times with the following parameters:

Reviewer: ```suggestion
`<InfiniteList>` will call `authProvider.canAccess()` 3 times with the following parameters:
```

For instance, given the following `<InfiniteList>`:

```tsx
import { InfiniteList, Datagrid, TextField } from 'react-admin';

// Resource name is "posts"
const PostList = () => (
    <InfiniteList>
        <Datagrid>
            <TextField source="title" />
            <TextField source="author" />
            <TextField source="published_at" />
        </Datagrid>
    </InfiniteList>
);
```

The `<InfiniteList>` will call `authProvider.canAccess()` using the following parameters:

- `{ action: "list", resource: "posts" }`.

Users without access will see [the `unauthorized` component](#unauthorized) instead of the list view.

**Tip**: If you want to control what columns users can see in the `<Datagrid>`, leverage [its access control feature](./Datagrid.md#access-control).

### Exported Fields

The `<ExportButton>` only exports fields the current user has the right to export (`{ action: "export", resource: "[resourceName].[fieldName]" }`).

For instance, given the following record shape:

```json
{
    "id": 0,
    "title": "Never Gonna Give You Up",
    "author": "Rick Astley",
}
```

> When a user clicks on the `<ExportButtonW`, `<InfiniteList>` will call `authProvider.canAccess()` 3 times with the following parameters:

Author: ```suggestion
When a user clicks on the `<ExportButton>`, `<InfiniteList>` will call `authProvider.canAccess()` 3 times with the following parameters:
```
